superduperdb/base/datalayer.py

```diff
@@ -838,8 +838,16 @@ def _add(
             serialized['version'] = object.version
             serialized['dict']['version'] = object.version
 
-        self._create_children(object, serialized)
         self.metadata.create_component(serialized)
+        self._create_children(object, serialized)
+        if object.child_components:
+            self.metadata.update_object(
+                identifier=object.identifier,  # type: ignore
+                type_id=object.type_id,
+                version=object.version,
+                key='dict',
+                value=serialized['dict'],
+            )
```

Comment:
1. Run `_create_component` first
2. Run `_create_children`
3. update the child metadata if parent component